src/main/scala/mesosphere/marathon/api/v2/json/Formats.scala

```diff
@@ -349,7 +349,8 @@ trait ContainerFormats {
           }
         case _ =>
           if (`type` == ContainerInfo.Type.DOCKER) {
-            throw SerializationFailedException("docker must not be empty")
+            //TODO(janisz): Hack!!! Think about explicit way to validate exclusive container types.
```

Comment:
What about 
```
import play.api.data.validation.ValidationError
throw JsResultException(Seq((JsPath() \ "image", Seq(ValidationError("error.path.missing")))))
```

Are you sure that `/image` is the correct path for the error message? Wondering because the error occurs because the path `/container/docker` is missing when `type="DOCKER"`.